workflows_core/engine/abstract_engine.py

```diff
@@ -102,26 +109,45 @@ def iterate(
         self,
         filters: Optional[List[Filter]] = None,
         select_fields: Optional[List[str]] = None,
+        max_retries: int = 3,
     ):
         if filters is not None:
             filters += self._get_workflow_filter()
         else:
             filters = self._get_workflow_filter()
-        while True:
-            chunk = self._dataset.get_documents(
-                self._chunksize,
-                filters=filters if filters is not None else self._filters,
-                select_fields=select_fields
-                if select_fields is not None
-                else self._select_fields,
-                after_id=self._after_id,
-                worker_number=self.worker_number,
-            )
-            self._after_id = chunk["after_id"]
-            if not chunk["documents"]:
-                break
-            yield chunk["documents"]
 
-    def update_chunk(self, chunk: List[Document]):
+        retry_count = 0
+        while True:
+            try:
+                chunk = self._dataset.get_documents(
+                    self._chunksize,
+                    filters=filters if filters is not None else self._filters,
+                    select_fields=select_fields
```

Comment:
Please remove one-liners. Maintaining clean multi-line logic is easier than maintaining one-liners